src/journal-remote/journal-remote.c

```diff
@@ -30,6 +30,18 @@
 
 #define filename_escape(s) xescape((s), "/ ")
 
+#if HAVE_MICROHTTPD
+void MHDDaemonWrapper_free(MHDDaemonWrapper *d) {
+        if (!d)
+            return;
+
+        MHD_stop_daemon(d->daemon);
```

Comment:
the microhttp docs do not say what happens if you pass a NULL pointer to MHD_stop_daemon(). For MHDDaemonWrapper_free() to act as a destructor that is robust in regards to getting half-allocated objects passed in you need to add an `if (d->daemon)` first here hence.